### PR TITLE
virtualdj 2024,8124 (new cask)

### PR DESCRIPTION
**Pull request by zotanmew**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=pullrequests).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/v/virtualdj.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+cask "virtualdj" do
+  version "2024.b8124"
+  sha256 "fddcdd538b4b1a8a70b7cba07f1f1303c73a387eeae3e7600eccdf96c9ec2e08"
+
+  url "https://download.virtualdj.com/install_virtualdj_#{version.dots_to_underscores}_mac.pkg"
+  name "VirtualDJ"
+  desc "#1 Most Popular DJ Software"
```
Comment (bevanjkay) [inline Casks/v/virtualdj.rb:7]:
```suggestion
  desc "DJ Software"
```

File: Casks/v/virtualdj.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+cask "virtualdj" do
+  version "2024.b8124"
```
Comment (bevanjkay) [inline Casks/v/virtualdj.rb:2]:
```suggestion
  version "2024,8124"
```

Split the version number with a comma between the version and then the build number.

File: Casks/v/virtualdj.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  sha256 "fddcdd538b4b1a8a70b7cba07f1f1303c73a387eeae3e7600eccdf96c9ec2e08"
+
+  url "https://download.virtualdj.com/install_virtualdj_#{version.dots_to_underscores}_mac.pkg"
```
Comment (bevanjkay) [inline Casks/v/virtualdj.rb:5]:
```suggestion
  url "https://download.virtualdj.com/install_virtualdj_#{version.csv.first}_b#{version.csv.second}_mac.pkg"
```

Update url to match new `version` format.

File: Casks/v/virtualdj.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  livecheck do
+    url "https://virtualdj.com/download/mac"
+    regex(/install_virtualdj_?(\d+_b\d+)_mac\.pkg/i)
+    strategy :header_match do |headers, regex|
+      headers["location"].scan(regex).map do |match|
+        match[0].tr("_", ".").to_s
+      end
+    end
+  end
```
Comment (bevanjkay) [inline Casks/v/virtualdj.rb:18]:
```suggestion
  livecheck do
    url "https://virtualdj.com/download/mac"
    regex(/install[._-]virtualdj[._-]?(\d+(?:\.\d+)*)[._-]b(\d+)[._-]mac\.pkg/i)
    strategy :header_match do |headers, regex|
      match = headers["location"].match(regex)
      next if match.blank?

      "#{match[1]},#{match[2]}"
    end
  end
```

File: Casks/v/virtualdj.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  pkg "install_virtualdj_#{version.dots_to_underscores}_mac.pkg"
```
Comment (bevanjkay) [inline Casks/v/virtualdj.rb:20]:
```suggestion
  pkg "install_virtualdj_#{version.csv.first}_b#{version.csv.second}_mac.pkg"
```

Update to match new `version` format.

**Review by bevanjkay — CHANGES_REQUESTED** [review]
Thanks @zotanmew for submitting this.
I have made some changes in my review.

**Comment by zotanmew** [comment]
Thank you, should be all done.

File: Casks/v/virtualdj.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  name "VirtualDJ"
+  desc "DJ Software"
+  homepage "https://virtualdj.com/"
+
+  livecheck do
+    url "https://virtualdj.com/download/mac"
+    regex(/install[._-]virtualdj[._-]?(\d+(?:\.\d+)*)[._-]b(\d+)[._-]mac\.pkg/i)
+    strategy :header_match do |headers, regex|
+      match = headers["location"].match(regex)
+      next if match.blank?
```
Comment (bevanjkay) [inline Casks/v/virtualdj.rb:15]: Thanks @zotanmew, just need to add a new line underneath the `next if` guard clause.

**Comment by zotanmew** [comment]
Apologies, I'm not that familiar with ruby syntax.

**Review by bevanjkay — APPROVED** [review]

**Review by miccal — APPROVED** [review]

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**